### ISSUE TITLE
iptables arguments in wrong order

### ISSUE THREAD

**Issue by kraney** [saltstack/salt#21389]
--to-source is another argument that needs to appear after --jump. (For SNAT.) It's currently provided out of order.
```
          ID: dmz-out
    Function: iptables.append
      Result: False
     Comment: Failed to set iptables rule for dmz-out.
              Attempted rule was /sbin/iptables -t nat -A POSTROUTING  --to-source 10.0.0.148 --source 10.0.1.43/32 --out eth0 --jump SNAT for ipv4
     Started: 20:39:20.587739
    Duration: 11.565 ms
     Changes:
```

Given that iptables is somewhat pluggable with kernel modules, and order matters because arguments like --jump SNAT add new valid parameters, would it maybe make sense to allow the ordering in the state to be explicit? For example, something like
```yaml
  - proto:
    - arg: tcp
    - dest-port: 80
  - jump:
    - arg: SNAT
    - to-source: 10.0.0.1
```
would imply that dest-port should come after --proto, and to-source should come after --jump. Basically, the set of things inside a dict should come after the dictionary key. The 'arg' key is for the argument immediately following the key.

I don't know, just a thought. In any case, --to-source needs to move after -j SNAT.

**Comment by kraney** [comment]
Never mind, I see it's already addressed in 2015.2, and then a different way on develop

**Comment by rallytime** [comment]
Thanks for the report and for following up here, @kraney. Feel free to let us know if you run into problems on either of the newer branches that you've mentioned.
